### PR TITLE
Organization dashboard entry point for staff

### PR DESCRIPTION
**Pull request by marcospri**
Include a public_id parameter to access the dashboards when launched from the organization view in the admin pages

## Testing
- Go into the admin pages, find an organization with local active course
- Make sure you didn't add yourself as an admin to the local organization.
- Navigate thru all three levels. Get results back.

### CODE REVIEW

File: lms/static/scripts/frontend_apps/components/dashboard/DashboardActivityFilters.tsx [hunk 1]
```diff
@@ -29,11 +29,16 @@ export default function DashboardActivityFilters({
   const { dashboard } = useConfig(['dashboard']);
   const { routes } = dashboard;
 
-  const courses = useAPIFetch<{ courses: Course[] }>(routes.courses);
+  const courses = useAPIFetch<{ courses: Course[] }>(routes.courses, {
+    public_id: dashboard.organization_public_id || [],
```
Comment (acelaya) [inline lms/static/scripts/frontend_apps/components/dashboard/DashboardActivityFilters.tsx:33]: I was looking at the type definition in `useAPIFetch`, and I think it should allow keys with value `undefined`, which should be just ignored.

Right now it only allows `string` or `string[]`, so your `|| []` workaround should work.

File: lms/static/scripts/frontend_apps/components/dashboard/test/OrganizationActivity-test.js [hunk 2]
```diff
@@ -175,20 +175,23 @@ describe('OrganizationActivity', () => {
       h_userid: ['123', '456'],
       assignment_id: [],
       course_id: [],
+      public_id: [],
```
Comment (marcospri) [inline lms/static/scripts/frontend_apps/components/dashboard/test/OrganizationActivity-test.js:178]: @acelaya while this seems to pass coverage it doesn't test we pass public_id when it's present, should we parametrize one other test? Which one? How? :grin:

File: lms/static/scripts/frontend_apps/components/dashboard/AssignmentActivity.tsx [hunk 3]
```diff
@@ -30,6 +30,7 @@ export default function AssignmentActivity() {
   );
   const students = useAPIFetch<StudentsResponse>(routes.students_metrics, {
     assignment_id: assignmentId,
+    public_id: dashboard.organization_public_id || [],
```
Comment (acelaya) [inline lms/static/scripts/frontend_apps/components/dashboard/AssignmentActivity.tsx:33]: If you rebase on `main`, you should now be able to drop `|| []`.

See https://github.com/hypothesis/lms/pull/6488

File: lms/static/scripts/frontend_apps/components/dashboard/test/OrganizationActivity-test.js (hunk 2 above)
Comment (acelaya) [reply lms/static/scripts/frontend_apps/components/dashboard/test/OrganizationActivity-test.js:178]: I can write a test if you want

**Review by acelaya — COMMENTED** [review]
With the latest changes from https://github.com/hypothesis/lms/pull/6488 that I just merged, passing `public_id` to `useAPIFetch` should now be simpler.

Also, I can add a test that covers the case in which the value is set in the config.

File: lms/static/scripts/frontend_apps/components/dashboard/test/OrganizationActivity-test.js (hunk 2 above)
Comment (acelaya) [reply lms/static/scripts/frontend_apps/components/dashboard/test/OrganizationActivity-test.js:178]: There https://github.com/hypothesis/lms/pull/6491

Comment (marcospri) [reply lms/static/scripts/frontend_apps/components/dashboard/test/OrganizationActivity-test.js:178]: That's great, I'm going to just merge it, thanks.

**Review by acelaya — APPROVED** [review]
Looks good!

**Pull request merged**